### ISSUE TITLE
Update to Solidity v0.4.17

### ISSUE THREAD

**Issue by onetom** [trufflesuite/truffle-init-default#5]
Currently there are warnings thrown after a fresh `truffle init` using version Truffle 4.0.0-beta.2:

```
$ npm init
$ npm i truffle@beta
$ node_modules/.bin/truffle compile
Compiling ./contracts/ConvertLib.sol...
Compiling ./contracts/MetaCoin.sol...
Compiling ./contracts/Migrations.sol...

Compilation warnings encountered:

/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/ConvertLib.sol:4:2: Warning: No visibility specified. Defaulting to "public".
	function convert(uint amount,uint conversionRate) returns (uint convertedAmount)
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/MetaCoin.sol:15:2: Warning: No visibility specified. Defaulting to "public".
	function MetaCoin() {
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/MetaCoin.sol:19:2: Warning: No visibility specified. Defaulting to "public".
	function sendCoin(address receiver, uint amount) returns(bool sufficient) {
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/MetaCoin.sol:27:2: Warning: No visibility specified. Defaulting to "public".
	function getBalanceInEth(address addr) returns(uint){
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/MetaCoin.sol:31:2: Warning: No visibility specified. Defaulting to "public".
	function getBalance(address addr) returns(uint) {
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/Migrations.sol:11:3: Warning: No visibility specified. Defaulting to "public".
  function Migrations() {
  ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/Migrations.sol:15:3: Warning: No visibility specified. Defaulting to "public".
  function setCompleted(uint completed) restricted {
  ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/Migrations.sol:19:3: Warning: No visibility specified. Defaulting to "public".
  function upgrade(address new_address) restricted {
  ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/ConvertLib.sol:4:2: Warning: Function state mutability can be restricted to pure
	function convert(uint amount,uint conversionRate) returns (uint convertedAmount)
 ^
Spanning multiple lines.
,/Users/onetom/gitlab.com/oax/asset-gateway-poc/00-skeleton/chain/contracts/MetaCoin.sol:31:2: Warning: Function state mutability can be restricted to view
	function getBalance(address addr) returns(uint) {
 ^
Spanning multiple lines.

Writing artifacts to ./build/contracts
```

These could be avoided by following the changes in the Solidity language mentioned in its release notes:
https://github.com/ethereum/solidity/blob/develop/Changelog.md#0417-2017-09-21

Related issue: https://github.com/trufflesuite/truffle/issues/616

**Comment by davesag** [comment]
Could we update to `Solidity v0.4.19` which appears to be the most recent now?

**Comment by vicnaum** [comment]
Should we update to `Solidity v0.4.23`? Cause now there's also a warning about the constructor() function.

**Comment by davesag** [comment]
The linters all currently break on the new syntax unfortunately.  Also solcov doesn't play nicely with the new constructor syntax.
